pkg/oneinch/fusionorder/encode.go

```diff
@@ -0,0 +1,22 @@
+package fusionorder
+
+import "math/big"
+
+// padOrTrim returns (size) bytes from input (bb)
+// Short bb gets zeros prefixed, Long bb gets left/MSB bits trimmed.
+func padOrTrim(bb []byte, size int) []byte {
+	l := len(bb)
+	if l == size {
+		return bb
+	}
+	if l > size {
+		return bb[l-size:]
+	}
+	tmp := make([]byte, size)
+	copy(tmp[size-l:], bb)
+	return tmp
+}
+
+func encodeInt64ToBytes(n int64, size int) []byte {
```

Comment:
take a look at `math.PaddedBigBytes` and `math.U256Bytes`, it may provided same function

Reply:
I have refactored to use `math.PaddingBigBytes` function